### PR TITLE
Fixed mp3 frame not found.

### PR DESCRIPTION
**Pull request by Krispeckt**
The error was caused by selecting the first track without checking the codec.

### CODE REVIEW

**Comment by topi314** [comment]
ah that makes sense, lavaplayer can also play aac tracks fyi

File: main/src/main/java/com/github/topi314/lavasrc/yandexmusic/YandexMusicAudioTrack.java [hunk 1]
```diff
@@ -51,11 +52,17 @@ private String getDownloadURL(String id) throws IOException, NoSuchAlgorithmExce
...
-		var downloadInfo = this.sourceManager.getDownloadStrings(downloadInfoLink);
-		if (downloadInfo == null) {
-			throw new IllegalStateException("No download URL found for track " + id);
+		var mp3ItemUrl = json.get("result")
+				.values()
+				.stream()
+				.filter(c -> c.get("codec").text().equals("mp3"))
+				.max(Comparator.comparingInt(b -> b.get("bitrateInKbps").as(Integer.class)))
+				.map(d -> d.get("downloadInfoUrl").text())
+				.orElse(null);
```
Comment (topi314) [inline main/src/main/java/com/github/topi314/lavasrc/yandexmusic/YandexMusicAudioTrack.java:61]: you can use `orElseThrow` here

File: main/src/main/java/com/github/topi314/lavasrc/yandexmusic/YandexMusicAudioTrack.java [hunk 2]
```diff
@@ -51,11 +52,17 @@ private String getDownloadURL(String id) throws IOException, NoSuchAlgorithmExce
...
 
-		var downloadInfoLink = json.get("result").values().get(0).get("downloadInfoUrl").text();
-		var downloadInfo = this.sourceManager.getDownloadStrings(downloadInfoLink);
-		if (downloadInfo == null) {
-			throw new IllegalStateException("No download URL found for track " + id);
+		var mp3ItemUrl = json.get("result")
+				.values()
+				.stream()
+				.filter(c -> c.get("codec").text().equals("mp3"))
+				.max(Comparator.comparingInt(b -> b.get("bitrateInKbps").as(Integer.class)))
```
Comment (topi314) [inline main/src/main/java/com/github/topi314/lavasrc/yandexmusic/YandexMusicAudioTrack.java:59]: in general always use longs
```suggestion
				.max(Comparator.comparingLong(b -> b.get("bitrateInKbps").asLong(0)))
```

File: main/src/main/java/com/github/topi314/lavasrc/yandexmusic/YandexMusicAudioTrack.java (hunk 1 above)
Comment (Krispeckt) [reply main/src/main/java/com/github/topi314/lavasrc/yandexmusic/YandexMusicAudioTrack.java:61]: I'll do a comit when I get home, didn't know about orElseThrow. :>

Comment (Krispeckt) [reply main/src/main/java/com/github/topi314/lavasrc/yandexmusic/YandexMusicAudioTrack.java:61]: > you can use `orElseThrow` here

Do I fix it or don't I need to fix it?

Comment (topi314) [reply main/src/main/java/com/github/topi314/lavasrc/yandexmusic/YandexMusicAudioTrack.java:61]: ideally you do

Comment (Krispeckt) [reply main/src/main/java/com/github/topi314/lavasrc/yandexmusic/YandexMusicAudioTrack.java:61]: Np

**Comment by Krispeckt** [comment]
> ah that makes sense, lavaplayer can also play aac tracks fyi

Yes, it makes sense when checking for codec type. But there is no codec type check in the code, Mp3Track is used right away.
See image

**Comment by topi314** [comment]
feel free to add support for this while you are at it

**Comment by Krispeckt** [comment]
> feel free to add support for this if you want, else this is ready to merge

Doesn't make sense as yandex always gives mp3.

**Review by topi314 — APPROVED** [review]

**Comment by topi314** [comment]
thanks

**Pull request merged**